cirq-google/cirq_google/transformers/analytical_decompositions/two_qubit_to_sycamore.py

```diff
@@ -0,0 +1,464 @@
+
+    Returns:
+        Four 2x2 unitaries. The first two are pre-rotations and last two are post rotations.
+    """
+    kak_u1 = cirq.kak_decomposition(target_unitary)
+    kak_u2 = cirq.kak_decomposition(source_unitary)
+
+    u_0 = kak_u1.single_qubit_operations_after[0] @ kak_u2.single_qubit_operations_after[0].conj().T
+    u_1 = kak_u1.single_qubit_operations_after[1] @ kak_u2.single_qubit_operations_after[1].conj().T
+
+    v_0 = (
+        kak_u2.single_qubit_operations_before[0].conj().T @ kak_u1.single_qubit_operations_before[0]
+    )
+    v_1 = (
+        kak_u2.single_qubit_operations_before[1].conj().T @ kak_u1.single_qubit_operations_before[1]
+    )
+
+    return v_0, v_1, u_0, u_1
+
+
+def _create_corrected_circuit(
+    target_unitary: np.ndarray, program: cirq.Circuit, q0: cirq.Qid, q1: cirq.Qid
+) -> cirq.OP_TREE:
+    """Adds pre/post single qubit rotations to `program` to make it equivalent to `target_unitary`.
+
+    Adds single qubit correction terms to the given circuit on 2 qubit s.t. it implements
+    `target_unitary`. This assumes that `program` implements a 2q unitary effect which has same
+    interaction coefficients as `target_unitary` in it's KAK decomposition and differs only in
+    local unitary rotations.
+
+    Args:
+        target_unitary: The unitary that should be implemented by the transformed `program`.
+        program: `cirq.Circuit` to be transformed.
+        q0: First qubit to operate on.
+        q1: Second qubit to operate on.
+
+    Yields:
+        Operations in `program` with pre and post rotations added s.t. the resulting `cirq.OP_TREE`
+        implements `target_unitary`.
+    """
+    # Get the local equivalents
+    b_0, b_1, a_0, a_1 = _find_local_equivalents(
+        target_unitary, program.unitary(qubit_order=cirq.QubitOrder.explicit([q0, q1]))
+    )
+
+    # Apply initial corrections
+    yield from _phased_x_z_ops(b_0, q0)
+    yield from _phased_x_z_ops(b_1, q1)
+
+    # Apply interaction part
+    yield program
+
+    # Apply final corrections
+    yield from _phased_x_z_ops(a_0, q0)
+    yield from _phased_x_z_ops(a_1, q1)
+
+
+def _phased_x_z_ops(mat: np.ndarray, q: cirq.Qid) -> Iterator[cirq.Operation]:
+    """Yields `cirq.PhasedXZGate` operation implementing `mat` if it is not identity."""
+    gate = cirq.single_qubit_matrix_to_phxz(mat)
+    if gate:
+        yield gate(q)
+
+
+def _rzz(theta: float, q0: cirq.Qid, q1: cirq.Qid) -> cirq.OP_TREE:
+    """Implements the Rzz Ising coupling gate (i.e. exp(-1j * theta * zz)) using Sycamore gates.
+
+    Args:
+        theta: The rotation parameter of Rzz Ising coupling gate.
+        q0: First qubit to operate on
+        q1: Second qubit to operate on
+
+    Yields:
+        The `cirq.OP_TREE` that implements the Rzz Ising coupling gate using Sycamore gates.
+    """
+    phi = -np.pi / 24
+    c_phi = np.cos(2 * phi)
+    target_unitary = scipy.linalg.expm(-1j * theta * UNITARY_ZZ)
```

Comment:
Could we make this simpler by just doing cirq.unitary(cirq.ZZ * theta) ?

Reply:
Indeed. Simplified and removed the global constants.